cartridge-cli.lua

```diff
@@ -420,12 +420,12 @@ end
 
 local function normalize_version(str)
     local patterns = {
-        "(%d+)%.(%d+)%.(%d+)-(%d+)-(%g+)",
-        "(%d+)%.(%d+)%.(%d+)-(%d+)",
-        "(%d+)%.(%d+)%.(%d+)-(%g+)",
-        "(%d+)%.(%d+)%.(%d+)",
-        "(%d+)%.(%d+)",
-        "(%d+)"
+        "^(%d+)%.(%d+)%.(%d+)%-(%d+)%-(g%g+)$",
```

Comment:
to be completely fair, git hash has constant length - (e.g. `g1b9031f2`) g + 8 symbols after